src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/NonInteractive/WsTrustResponse.cs

```diff
@@ -119,29 +127,31 @@ internal static WsTrustResponse CreateFromResponseDocument(XDocument responseDoc
                     }
                 }
 
-                if (parseResponse)
+                if (!parseResponse)
+                {
+                    throw new AdalException(AdalError.ParsingWsTrustResponseFailed);
+                }
+
+                IEnumerable<XElement> tokenResponses =
+                    responseDocument.Descendants(t + "RequestSecurityTokenResponse");
+                foreach (var tokenResponse in tokenResponses)
                 {
-                    IEnumerable<XElement> tokenResponses =
-                        responseDocument.Descendants(t + "RequestSecurityTokenResponse");
-                    foreach (var tokenResponse in tokenResponses)
+                    XElement tokenTypeElement = tokenResponse.Elements(t + "TokenType").FirstOrDefault();
+                    if (tokenTypeElement == null)
                     {
-                        XElement tokenTypeElement = tokenResponse.Elements(t + "TokenType").FirstOrDefault();
-                        if (tokenTypeElement == null)
-                        {
-                            continue;
-                        }
-
-                        XElement requestedSecurityToken =
-                            tokenResponse.Elements(t + "RequestedSecurityToken").FirstOrDefault();
-                        if (requestedSecurityToken == null)
-                        {
-                            continue;
-                        }
-
-                        // TODO #123622: We need to disable formatting due to a potential service bug. Remove the ToString argument when problem is fixed.
-                        tokenResponseDictionary.Add(tokenTypeElement.Value,
-                            requestedSecurityToken.FirstNode.ToString(SaveOptions.DisableFormatting));
+                        continue;
                     }
+
+                    XElement requestedSecurityToken =
+                        tokenResponse.Elements(t + "RequestedSecurityToken").FirstOrDefault();
+                    if (requestedSecurityToken == null)
+                    {
+                        continue;
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS1135)